modules/javafx.graphics/src/main/java/javafx/scene/image/PixelBuffer.java

```diff
@@ -0,0 +1,220 @@
+
+/**
+ * The {@code PixelBuffer} class represents pixel data that is constructed from
+ * a {@link Buffer java.nio.Buffer} supplied by the application.
+ * A {@link WritableImage} can use this {@code PixelBuffer} directly without copying the pixel data.
+ * This {@code PixelBuffer} can be shared among multiple {@code WritableImage}s.
+ * Pixel data should be stored either in an {@link IntBuffer} using a {@link PixelFormat} of type
+ * {@code INT_ARGB_PRE} or in a {@link ByteBuffer} using a {@link PixelFormat} of type {@code BYTE_BGRA_PRE}.
+ * When the {@code Buffer} is updated using the {@link #updateBuffer PixelBuffer.updateBuffer} method,
+ * all {@code WritableImage}s that were created using this {@code PixelBuffer} are redrawn.
+ * <p>
+ * Example code that shows how to create a {@code PixelBuffer}:
+ * <pre>{@code
+ * // Creating a PixelBuffer using BYTE_BGRA_PRE pixel format.
+ * ByteBuffer byteBuffer = ByteBuffer.allocateDirect(width * height * 4);
+ * PixelFormat<ByteBuffer> pixelFormat = PixelFormat.getByteBgraPreInstance();
+ * PixelBuffer<ByteBuffer> pixelBuffer = new PixelBuffer<>(width, height, byteBuffer, pixelFormat);
+ * Image img = new WritableImage(pixelBuffer);
+ *
+ * // Creating a PixelBuffer using INT_ARGB_PRE pixel format.
+ * IntBuffer intBuffer = IntBuffer.allocate(width * height);
+ * PixelFormat<IntBuffer> pixelFormat = PixelFormat.getIntArgbPreInstance();
+ * PixelBuffer<IntBuffer> pixelBuffer = new PixelBuffer<>(width, height, intBuffer, pixelFormat);
+ * Image img = new WritableImage(pixelBuffer);
+ * }</pre>
+ *
+ * @param <T> the type of {@link Buffer} that should be either {@link IntBuffer} or {@link ByteBuffer}
+ * @see WritableImage#WritableImage(PixelBuffer)
+ * @since 13
+ */
+public class PixelBuffer<T extends Buffer> {
+
+    private final T buffer;
+    private final int width;
+    private final int height;
+    private final PixelFormat<T> pixelFormat;
+    private final List<WeakReference<WritableImage>> imageRefs;
+
+    /**
+     * Constructs a {@code PixelBuffer} using the specified {@code Buffer} and {@code PixelFormat}.
+     * The type of the specified {@code PixelFormat} must be either {@code PixelFormat.Type.INT_ARGB_PRE}
+     * or {@code PixelFormat.Type.BYTE_BGRA_PRE}.
+     * The constructor does not allocate memory to store the pixel data. The application must provide
+     * a buffer with sufficient memory for the combination of dimensions {@code (width, height)} and
+     * {@code PixelFormat}. The {@code PixelFormat.Type.INT_ARGB_PRE} requires an {@link IntBuffer} with
+     * minimum capacity of {@code width * height} and the {@code PixelFormat.Type.BYTE_BGRA_PRE} requires
+     * a {@link ByteBuffer} with minimum capacity of {@code width * height * 4}.
+     *
+     * @param width       width in pixels of this {@code PixelBuffer}
+     * @param height      height in pixels of this {@code PixelBuffer}
+     * @param buffer      {@code java.nio.Buffer} that stores the pixel data
```

Comment:
I think that just "the buffer that..." is good since its type is already defined.